implementation/.github/workflows/approve-bot-pr.yml

```diff
@@ -0,0 +1,73 @@
+name: Approve Bot PRs
+
+on:
+  workflow_run:
+    workflows: ["Test Pull Request"]
+    types:
+      - completed
+
+jobs:
+  download:
+    name: Download PR Artifact
+    if: >
+      ${{ github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "event-payload"
+            })[0];
+            var download = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/event-payload.zip', Buffer.from(download.data));
+      # test-pull-request event-payload will be unzipped into github.workspace
+      # See https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#webhook-payload-example-32
+      - run: unzip event-payload.zip
```

Comment:
I was hoping to avoid inlining a bunch of gnarly javascript into our workflow for this. Were you planning to pull this out into an action, maybe rewritten in `bash` so that its a bit easier for the team to maintain?

Reply:
That wasn't my original intent, but that definitely makes sense and is something I will do